### ISSUE TITLE
SSPullToRefresh does not work correctly on ios11 

### ISSUE THREAD

**Issue by avshiyanov** [soffes/SSPullToRefresh#65]

**Comment by enzosterro** [comment]
@avshiyanov, could you provide more details, please?

**Comment by avshiyanov** [comment]
@enzosterro I think this problems appears only after making a new build project in new xCode 9, this problem is related to content inset of UITableView and pull to refresh looks wired - for some cases it's under navigation bar and don't disappears after making "pull to refresh"

**Comment by enzosterro** [comment]
@avshiyanov, it's interesting. I've got no problems that you've described.

**Comment by avshiyanov** [comment]
<img width="382" alt="layout" src="https://user-images.githubusercontent.com/4908325/31181125-3108d06c-a929-11e7-9377-74ac43ffb730.png">
@enzosterro It happen after building project on xCode9

**Comment by avshiyanov** [comment]
@enzosterro I think problem inside of SSPullToRefresh, I tried another pulls to refresh and it works but with not similar UI . I think you need test this control under XCode 9 with different simulators and you will see that it's broken, please look on screenshot

**Comment by ScottRobbins** [comment]
I also have the same issue. Haven't had time to dig into the code to see why.

**Comment by avshiyanov** [comment]
@ScottRobbins Yes it's contentInset issue, I was able to fix on client, but think best place to fix it in the library @enzosterro . Not sure if best solution is hardcode top inset for iPhoneX, but it works for me for all iOS and deviceTypes

    UIEdgeInsets inset = self.tableView.contentInset;
    if (@available(iOS 11.0, *)) {
        inset = self.tableView.safeAreaInsets;
    } else {
        inset = self.tableView.contentInset;
    }
    inset.top = isDevicePhoneX ? 24 : 0;
    
    self.pullToRefreshView.defaultContentInset = inset;
    
    if (@available(iOS 11.0, *)) {
        self.tableView.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentNever;
    } else {
        self.automaticallyAdjustsScrollViewInsets = NO;
    }

**Comment by enzosterro** [comment]
@avshiyanov, sorry, but it's not mine library. And yes, after I've built project in xCode 9, I've stuck with same problems.

**Comment by avshiyanov** [comment]
@enzosterro I see

**Comment by avshiyanov** [comment]
@ScottRobbins Do you have solution for fix?

**Comment by pjechris** [comment]
I do have kind of the same issue, but on Xcode8/iOS 10.
After a pull to refresh top of my tableview is hidden. Seems like contentInset is miscalculated.
My hierarchy: NavigationController -> Tabbarcontroller -> UIViewController -> UIView -> UITableView

**Comment by avshiyanov** [comment]
@pjechris Hi, I fixed it at https://github.com/avshiyanov/SSPullToRefresh
